Data_Structures_Questions.md

```diff
@@ -37,23 +37,34 @@ Answer the following questions for each of the data structures you implemented a
 ## Doubly Linked List
 
 1. What is the runtime complexity of `ListNode.insert_after`?
+    - O(1)
 
 2. What is the runtime complexity of `ListNode.insert_before`?
+    - O(1)
 
 3. What is the runtime complexity of `ListNode.delete`?
+    - O(1)
 
 4. What is the runtime complexity of `DoublyLinkedList.add_to_head`?
+    - O(1)
 
 5. What is the runtime complexity of `DoublyLinkedList.remove_from_head`?
+    - O(1)
 
 6. What is the runtime complexity of `DoublyLinkedList.add_to_tail`?
+    - O(1)
 
 7. What is the runtime complexity of `DoublyLinkedList.remove_from_tail`?
+    - O(1)
 
 8. What is the runtime complexity of `DoublyLinkedList.move_to_front`?
+    - O(1)
 
 9. What is the runtime complexity of `DoublyLinkedList.move_to_end`?
+    - O(1)
 
 10. What is the runtime complexity of `DoublyLinkedList.delete`?
+    - O(1)
 
-    a. Compare the runtime of the doubly linked list's `delete` method with the worst-case runtime of the JS `Array.splice` method. Which method generally performs better?
\ No newline at end of file
+    a. Compare the runtime of the doubly linked list's `delete` method with the worst-case runtime of the JS `Array.splice` method. Which method generally performs better?
+            - I don't know JS, but deleting a value from the middle of an array would be O(n) whereas dll delete() would only be O(1). Deleting from the middle of an array causes each of the succeeding values in the array to move up one spot in memory. With dlls, it doesn't matter where the each of the value is in memory, just that the head and tail of the item to delete point at each other.
```

Comment:
:heavy_check_mark: 
yep exactly this explanation is perfect, arrays need their elements to be adjacent to one another in blocks of memory.

Reply:
And I'll take note of this question not being designed with your track in mind, great job referencing js nonetheless. As you can see arrays are pretty much the same in any language, minor details here and there.